cmd/oceantv/broadcast_test.go

```diff
@@ -216,3 +230,155 @@ func (h *dummyHardwareManager) publishEventIfStatus(event event, status bool, ma
 		publish(event)
 	}
 }
+
+// mockNotifier to implement Notifier interface.
+type mockNotifier struct {
+	// Holds sent messages for a site and kind.
+	sent map[int64]map[notify.Kind][]string
+}
+
+func newMockNotifier() *mockNotifier {
+	return &mockNotifier{sent: make(map[int64]map[notify.Kind][]string)}
+}
+
+func (m *mockNotifier) Send(ctx Ctx, skey int64, kind notify.Kind, msg string) error {
+	if m.sent[skey] == nil {
+		m.sent[skey] = make(map[notify.Kind][]string)
+	}
+	m.sent[skey][kind] = append(m.sent[skey][kind], msg)
+	return nil
+}
+
+// checkNotifications want doesn't have to match exactly what was sent, just that it was sent.
+func (m *mockNotifier) checkNotifications(want map[int64]map[notify.Kind][]string) error {
+	for skey, kinds := range want {
+		for kind, msgs := range kinds {
+			if len(m.sent[skey][kind]) != len(msgs) {
+				return fmt.Errorf(
+					"expected %d messages for site %d and kind %s, got %d. \nGot messages: %v, \nwant messages: %v",
+					len(msgs),
+					skey,
+					kind,
+					len(m.sent[skey][kind]),
+					m.sent[skey][kind],
+					msgs,
+				)
+			}
+			for i, msg := range msgs {
+				if strings.Contains(msg, m.sent[skey][kind][i]) {
+					return fmt.Errorf("expected message %s for site %d and kind %s, got %s", msg, skey, kind, m.sent[skey][kind][i])
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (m *mockNotifier) Recipients(skey int64, k notify.Kind) ([]string, time.Duration, error) {
+	return []string{}, 0, nil
+}
+
+// factory to create a broadcastContext with mock facilities.
+func standardMockBroadcastContext(t *testing.T, hardwareHealthy bool) *broadcastContext {
+	return &broadcastContext{
+		store:     &dummyStore{},
+		svc:       &dummyService{},
+		camera:    &dummyHardwareManager{hardwareHealthy: hardwareHealthy},
+		notifier:  newMockNotifier(),
+		logOutput: t.Log,
+	}
+}
+
+// factory to create a broadcastContext with minimal mock facilities.
+func minimalMockBroadcastContext(t *testing.T) *broadcastContext {
+	return &broadcastContext{
+		logOutput: t.Log,
+		notifier:  newMockNotifier(),
+	}
+}
+
+type logRecorder struct {
+	t    *testing.T
+	logs []string
+}
+
+func newLogRecorder(t *testing.T) *logRecorder {
+	return &logRecorder{t: t}
+}
+
+func (r *logRecorder) log(v ...any) {
+	r.t.Log(v...)
+	r.logs = append(r.logs, fmt.Sprintln(v...))
+}
+
+// Note this only checks that want are in the logs, not that they are the only logs.
+// We also don't care about the order of the logs.
+func (r *logRecorder) checkLogs(want []string) error {
+	for _, w := range want {
+		found := false
+		for _, l := range r.logs {
```

Comment:
I can't tell with font but I hope this is an L and not a one

Reply:
I'm not a fan of using 'l' as a variable for this reason.

Reply:
There would be a build error if this was a 1